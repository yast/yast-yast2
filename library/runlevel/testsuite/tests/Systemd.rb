# encoding: utf-8

# ***************************************************************************
#
# Copyright (c) 2002 - 2012 Novell, Inc.
# All Rights Reserved.
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of version 2 of the GNU General Public License as
# published by the Free Software Foundation.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, contact Novell, Inc.
#
# To contact Novell about this file by physical or electronic mail,
# you may find current contact information at www.novell.com
#
# ***************************************************************************
module Yast
  class SystemdClient < Client
    def main
      Yast.import "Systemd"
      Yast.include self, "testsuite.rb"

      DUMP("Systemd::Running() tests")

      # Note: \0 should be used here as the separator, but YCP does not allows that...

      # test standard System V init
      TEST(->() { Systemd.Running }, [{ "target" => { "stat" => {} } }, {}, {}], nil)

      # systemd
      TEST(
        ->() { Systemd.Running },
        [{ "target" => { "stat" => { "isdir" => true } } }, {}, {}],
        nil
      )

      DUMP("Systemd::SetDefaultRunlevel() tests")

      # test invalid parameters
      TEST(->() { Systemd.SetDefaultRunlevel(nil) },
        [
          {},
          {},
          { "target" => { "bash" => 0 } }
        ], nil)

      TEST(->() { Systemd.SetDefaultRunlevel(-1) },
        [
          {},
          {},
          { "target" => { "bash" => 0 } }
        ], nil)

      TEST(->() { Systemd.SetDefaultRunlevel(7) },
        [
          {},
          {},
          { "target" => { "bash" => 0 } }
        ], nil)

      # test valid parameters
      TEST(->() { Systemd.SetDefaultRunlevel(3) },
        [
          {},
          {},
          { "target" => { "bash" => 0 } }
        ], nil)

      TEST(->() { Systemd.SetDefaultRunlevel(5) },
        [
          {},
          {},
          { "target" => { "bash" => 0 } }
        ], nil)

      # test failure
      TEST(->() { Systemd.SetDefaultRunlevel(5) },
        [
          {},
          {},
          { "target" => { "bash" => 1 } }
        ], nil)

      DUMP("Systemd::DefaultRunlevel() tests")

      # test missing / invalid (not a symlink) default
      TEST(->() { Systemd.DefaultRunlevel },
        [
          { "target" => { "symlink" => nil } },
          {},
          {}
        ], nil)

      # test numeric runlevel
      TEST(
        ->() { Systemd.DefaultRunlevel },
        [
          {
            "target" => { "symlink" => "/lib/systemd/system/runlevel3.target" }
          },
          {},
          {}
        ],
        nil
      )

      # test symbolic runlevel
      TEST(
        ->() { Systemd.DefaultRunlevel },
        [
          {
            "target" => { "symlink" => "/lib/systemd/system/graphical.target" }
          },
          {},
          {}
        ],
        nil
      )

      # test unknown symbolic runlevel
      TEST(
        ->() { Systemd.DefaultRunlevel },
        [
          {
            "target" => {
              "symlink" => "/lib/systemd/system/unknown_default.target"
            }
          },
          {},
          {}
        ],
        nil
      )

      nil
    end
  end
end

Yast::SystemdClient.new.main
