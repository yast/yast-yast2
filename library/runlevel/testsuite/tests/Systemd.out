Dump	Systemd::Running() tests
Read	.target.stat "/sys/fs/cgroup/systemd" $[]
Return	false
Read	.target.stat "/sys/fs/cgroup/systemd" $["isdir":true]
Return	true
Dump	Systemd::SetDefaultRunlevel() tests
Log	Invalid default runlevel (must be in range 0..6): nil
Return	false
Log	Invalid default runlevel (must be in range 0..6): -1
Return	false
Log	Invalid default runlevel (must be in range 0..6): 7
Return	false
Execute	.target.bash "/bin/ln -s -f /usr/lib/systemd/system/runlevel3.target /etc/systemd/system/default.target" 0
Return	true
Execute	.target.bash "/bin/ln -s -f /usr/lib/systemd/system/runlevel5.target /etc/systemd/system/default.target" 0
Return	true
Execute	.target.bash "/bin/ln -s -f /usr/lib/systemd/system/runlevel5.target /etc/systemd/system/default.target" 1
Return	false
Dump	Systemd::DefaultRunlevel() tests
Read	.target.symlink "/etc/systemd/system/default.target" nil
Log	Cannot read symlink target of /etc/systemd/system/default.target
Return	nil
Read	.target.symlink "/etc/systemd/system/default.target" "/lib/systemd/system/runlevel3.target"
Return	3
Read	.target.symlink "/etc/systemd/system/default.target" "/lib/systemd/system/graphical.target"
Return	5
Read	.target.symlink "/etc/systemd/system/default.target" "/lib/systemd/system/unknown_default.target"
Return	nil
